pkg/api/validation/validation.go

```diff
@@ -628,6 +628,9 @@ func validateHostPathVolumeSource(hostPath *api.HostPathVolumeSource, fldPath *f
 	if len(hostPath.Path) == 0 {
 		allErrs = append(allErrs, field.Required(fldPath.Child("path"), ""))
 	}
+
+	allErrs = append(allErrs, validatePathNoBacksteps(hostPath.Path, fldPath.Child("path"))...)
```

Comment:
Neat: put it to `else` block to not being executed when path is empty. Or add a `return` to the `if` body.